clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/provider/agent/KubernetesServerGroupCachingAgent.groovy

```diff
@@ -42,7 +42,7 @@ class KubernetesServerGroupCachingAgent implements CachingAgent, OnDemandAgent,
   @Deprecated
   private static final String LEGACY_ON_DEMAND_TYPE = 'KubernetesCluster'
 
-  private static final String ON_DEMAND_TYPE = 'Cluster'
+  private static final List<String> ON_DEMAND_TYPES = ['ServerGroup', 'Cluster', 'Instance']
```

Comment:
The thing is this agent is responsible for server groups, clusters, and instances. When the top level `CachingAgent` class queries each `CachingAgent` subclass, it decides whether or not to force a cache refresh based on the response of `handle(type, name)`. Maybe @cfieber knows what the right thing is to do here.